central/productusage/datastore/securedunits/datastore_impl.go

```diff
@@ -18,26 +21,46 @@ var (
 )
 
 type dataStoreImpl struct {
+	store        store.Store
 	clusterStore clusterStoreI
 	cache        cache.Cache
 }
 
 var _ DataStore = (*dataStoreImpl)(nil)
 
 // Get returns the object, if it exists from the store.
-func (ds *dataStoreImpl) Get(ctx context.Context, _ *types.Timestamp, _ *types.Timestamp) (<-chan *storage.SecuredUnits, error) {
+func (ds *dataStoreImpl) Get(ctx context.Context, from *types.Timestamp, to *types.Timestamp) (<-chan *storage.SecuredUnits, error) {
 	if err := sac.VerifyAuthzOK(usageSAC.ReadAllowed(ctx)); err != nil {
 		return nil, errors.Wrap(err, "cannot permit to get usage data")
 	}
-	return nil, errors.New("not implemented")
+	if from == nil {
+		from, _ = types.TimestampProto(time.Time{})
+	}
+	if to == nil {
+		to = types.TimestampNow()
+	}
+	result := make(chan *storage.SecuredUnits)
+	go func() {
+		defer close(result)
+		_ = ds.store.Walk(ctx, func(record *storage.SecuredUnits) error {
```

Comment:
We should probably do something with the error from the `Walk` even if all we do is log it.